localstack/services/cloudformation/models/s3.py

```diff
@@ -157,18 +155,26 @@ def _put_bucket_versioning(resource_id, resources, resource_type, func, stack_na
                     },
                 )
 
+        def _create_bucket(resource_id, resources, resource_type, func, stack_name):
+            s3_client = aws_stack.connect_to_service("s3")
+            resource = resources[resource_id]
+            props = resource["Properties"]
+            bucket_name = props.get("BucketName")
+            try:
+                s3_client.head_bucket(Bucket=bucket_name)
+            except Exception:
```

Comment:
nit: too broad. we should probably explicitly check for it being a not found exception here and avoid swallowing any unrelated issues here

Reply:
Added a stronger check 👍